caiman/source_extraction/cnmf/lazy_arrays.py

```diff
@@ -0,0 +1,439 @@
+from warnings import warn
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import *
+from itertools import product as iter_product
+from time import time
+
+
+import numpy as np
+from scipy.sparse import csc_matrix
+import h5py
```

Comment:
If we can alphabetise the imports and add a #!/usr/bin/env python at the top to hint to editors that it's python